### PR TITLE
[DOC-550] Add hidden var on tilde commands 

### PR DESCRIPTION
**Pull request by dandimeo**
#### Upstream PRs

- 3.10:
- 3.11:
- 3.12:

### CODE REVIEW

File: toolchain/arangoproxy/internal/format/format.go [hunk 1]
```diff
@@ -15,16 +15,34 @@ import (
...
+		line = re.ReplaceAllString(line, "var ")
+
+		if strings.Contains(line, "~") {
+			if !strings.Contains(line, "var ") {
+				line = fmt.Sprintf("var hidden = %s", line)
+			}
+			line = strings.ReplaceAll(line, "~", "")
+		}
+
+		re = regexp.MustCompile(`(?m)}\n *catch`)
```
Comment (Simran-B) [inline toolchain/arangoproxy/internal/format/format.go:39]:
```suggestion
		re = regexp.MustCompile(`(?m)\}\n *catch`)
```

File: toolchain/arangoproxy/internal/format/format.go [hunk 2]
```diff
@@ -15,16 +15,34 @@ import (
...
-	code = re.ReplaceAllString(code, "} catch")
-	return code
+
+	lines := strings.Split(code, "\n")
+
+	for _, line := range lines {
+		re := regexp.MustCompile(`(?m)let |const `)
+		line = re.ReplaceAllString(line, "var ")
+
+		if strings.Contains(line, "~") {
```
Comment (Simran-B) [inline toolchain/arangoproxy/internal/format/format.go:32]: Use a regex for safety? `"^\s*~"`

File: toolchain/arangoproxy/internal/format/format.go [hunk 3]
```diff
@@ -15,16 +15,34 @@ import (
...
+			if !strings.Contains(line, "var ") {
+				line = fmt.Sprintf("var hidden = %s", line)
+			}
+			line = strings.ReplaceAll(line, "~", "")
```
Comment (Simran-B) [inline toolchain/arangoproxy/internal/format/format.go:36]: Also regex here? `"^(\s*)~"` and substitute with `"$1 "`

**Review by Simran-B — CHANGES_REQUESTED** [review]
Generally works but there seem to be additional cases:
https://deploy-preview-189--docs-hugo.netlify.app/3.11/aql/execution-and-performance/query-profiling/#example-simple-aql-query
The output should not contain (one of) the document insertions marked in yellow:
image

The line in question is:
```js
~for (let i=0; i < 10000; i++) { db.acollection.insert({value:i}); }
```

File: toolchain/arangoproxy/internal/format/format.go (hunk 2 above)
Comment (dandimeo) [reply toolchain/arangoproxy/internal/format/format.go]: added in last commit

File: toolchain/arangoproxy/internal/format/format.go (hunk 3 above)
Comment (dandimeo) [reply toolchain/arangoproxy/internal/format/format.go]: added in last commit

File: toolchain/arangoproxy/internal/format/format.go [hunk 4]
```diff
@@ -15,16 +15,34 @@ import (
...
-	code = re.ReplaceAllString(code, "} catch")
-	return code
+
+	lines := strings.Split(code, "\n")
+
+	for _, line := range lines {
+		re := regexp.MustCompile(`(?m)let |const `)
+		line = re.ReplaceAllString(line, "var ")
+
+		tildeRE := regexp.MustCompile(`(?m)^(\s*)~`)
```
[the remark this answers is not in the document]
Comment (dandimeo) [reply toolchain/arangoproxy/internal/format/format.go]: added in last commit

**Review by Simran-B — APPROVED** [review]
Corner case fixed, all other samples still fine!

**Pull request merged**